lib/diagnostic.js

```diff
@@ -5,29 +5,54 @@
 // and returns the minimum from that set
 // assume that all the argument are numbers
 const min = function min() {
-
+let mini = 0;
+for (let i = 0; i < arguments.length; i++) {
+  if(mini > arguments[i]) {
+    mini = arguments[i];
+  }
+}
+return mini;
 };
 
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
 const sum = function sum() {
-
+  let sum = 0;
+  for (let i = 0; i < arguments.length; i++) {
+      sum += arguments[i];
+  }
+  return sum;
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
 const newArray = function newArray(length, defaultValue) {
-
+  let newArr = [];
+  let i = 0;
+  while (i < length){
+    newArr.push(defaultValue);
+    i++;
+  }
+  return newArr;
 };
 
 // Question 4
 // Add code to newArray so that if defaultValue is a
 // function then the elements in the array will be set
 // to the result of invoking defaultValue with the
 // element's index as an argument
+const newArray = function newArray(length, defaultValue) {
```

Comment:
Hey Brian,


Thank you for your grading. I changed my function name, but it still won't work. I got "

Error: Expected [ Function, Function, Function ] to equal [ 3, 2, 1 ]." error when I run "grunt test". Could you let me know what is wrong with my code? Thank you.


const newArrayNew = function newArrayNew(length, defaultValue) {
  let newArr = [];
  let i = 0;
  while (i < length) {
    newArr[i] = defaultValue(i); //this is the problem, but I don't know why
    i++;
  }
  return newArr;
};



________________________________
From: Brian Berzellini <notifications@github.com>
Sent: Tuesday, May 24, 2016 6:20 PM
To: ga-wdi-boston/js-functions-ins-and-outs-diagnostic
Cc: tina513; Author
Subject: Re: [ga-wdi-boston/js-functions-ins-and-outs-diagnostic] Answer for js-functions-ins-and-outs-diagnostic (#88)


In lib/diagnostic.js<https://github.com/ga-wdi-boston/js-functions-ins-and-outs-diagnostic/pull/88#discussion_r64485216>:

>  };
>
>  // Question 4
>  // Add code to newArray so that if defaultValue is a
>  // function then the elements in the array will be set
>  // to the result of invoking defaultValue with the
>  // element's index as an argument
> +const newArray = function newArray(length, defaultValue) {


You cannot reassign newArray since it is already defined above.

-
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/ga-wdi-boston/js-functions-ins-and-outs-diagnostic/pull/88/files/5ffa7271db6943178ac91ae75ca1448d03ac10a0#r64485216>